backend/src/apiserver/resource/resource_manager.go

```diff
@@ -535,6 +535,18 @@ func (r *ResourceManager) CreateJob(apiJob *api.Job) (*model.Job, error) {
 			},
 		},
 	}
+
+	// Marking auto-added artifacts as optional. Otherwise most older workflows will start failing after upgrade to Argo 2.3.
+	// TODO: Fix the components to explicitly declare the artifacts they really output.
+	// TODO: Change the compiler to stop auto-adding those two atrifacts to all tasks.
```

Comment:
IIRC the compiler no longer adds those two artifact to all tasks. @Ark-kun can you correct me if I'm wrong?

Reply:
Yes. The compiler stopped doing this in SDK v0.1.29: https://github.com/kubeflow/pipelines/pull/2046.
Pipelines compiled by older SDKs have the artifacts hardcoded in them.

Reply:
Does this todo still apply?

Reply:
I believe no. That's why I was asking :)

Reply:
Great to know. I'll remove the complier TODO, and also in the compiler TODO in the one-time run.